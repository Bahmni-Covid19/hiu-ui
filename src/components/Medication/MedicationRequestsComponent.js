import React from 'react';
import * as PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import { formatDateString } from '../common/HealthInfo/FhirResourcesUtils';
import TableStyles from '../common/Styles/Table.style';
import MedicationPriority from './MedicationPriority';
import MedicationNote from './MedicationNote';
import MedicationDose from './MedicationDose';


const findMedicationName = (mr) => {
  const medicationConcept 
    = (mr.medicationReference && mr.medicationReference.targetResource) 
      ? mr.medicationReference.targetResource.code 
      : mr.medicationCodeableConcept;
  if (!medicationConcept) return "Unspecified";
  if (medicationConcept.text) {
    return medicationConcept.text;
  }
  if (medicationConcept.coding) {
    return medicationConcept.coding[0].display
      ? medicationConcept.coding[0].display
      : medicationConcept.coding[0].code;
  }
  return 'Unspecified';
  
};

// eslint-disable-next-line max-len
const MedicationRequestsComponent = ({ medicationRequests }) => (medicationRequests && medicationRequests.length > 0 ? (
  <TableStyles>
    <TableContainer className="table-container" component={Paper}>
      <Table className="table" aria-label="simple table">
        <TableHead>
          <TableRow className="table-head">
            <TableCell className="header" colSpan={4}>
              Medication:
            </TableCell>
          </TableRow>
          <TableRow className="table-head">
            <TableCell align="left">Date</TableCell>
            <TableCell align="left">Medication</TableCell>
            <TableCell align="left">Dosing Instruction</TableCell>
            <TableCell align="left">Additional Info</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {medicationRequests.map((mr) => (
            <TableRow key={mr.authoredOn}>
              <TableCell className="table-cell">
                {mr.authoredOn ? formatDateString(mr.authoredOn) : ''}
              </TableCell>
              <TableCell className="table-cell">
                {findMedicationName(mr)}
                {` (${mr.status})`}
              </TableCell>
              <TableCell className="table-cell">
                <MedicationDose dosageInstructions={mr.dosageInstruction} />
              </TableCell>
              <TableCell className="table-cell">
                <ul className="mediation-list-item">
                  {<MedicationPriority mr={mr} />}
                  {<MedicationNote mr={mr} />}
                </ul>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  </TableStyles>
) : (
  <div />
));

const medicationRequestShape = {
  status: PropTypes.string,
  authoredOn: PropTypes.string,
  dosageInstruction: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
  })),
  medicationReference: PropTypes.shape({
    targetResource: PropTypes.shape({
      code: PropTypes.shape({
        coding: PropTypes.arrayOf(PropTypes.shape({
          code: PropTypes.string,
          display: PropTypes.string,
        })),
      }),
    }),
  }),
};

MedicationRequestsComponent.propTypes = {
  medicationRequests: PropTypes.arrayOf(PropTypes.shape(medicationRequestShape)),
};

MedicationRequestsComponent.defaultProps = {
  medicationRequests: [],
};

export default MedicationRequestsComponent;
