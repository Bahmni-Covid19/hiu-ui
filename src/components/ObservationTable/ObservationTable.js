import React, { useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import {
  generateObservableEntity,
  generateObservableEntityValue,
  generateObservableEntityStatusInterpretation
} from "./ObservationTableHelperFunctions";
import ResourceData from "./ResourceData";

const useStyles = makeStyles({
  table: {
    minWidth: 650
  },
  tableHead: {
    backgroundColor: "#c8c8c8"
  },
  evenTableRow: {
    backgroundColor: "#f3f3f3"
  }
});

const ObservationTable = ({ healthInfo }) => {
  const classes = useStyles();

  console.log("*healthInfo*", healthInfo);

  const Observation = ResourceData.entry.find(
    item => item.resource.resourceType === "Observation"
  ).resource;

  return (
    <TableContainer component={Paper}>
      <Table className={classes.table} aria-label="simple table">
        <TableHead>
          <TableRow className={classes.tableHead}>
            <TableCell align="left">Date</TableCell>
            <TableCell align="left">Observation</TableCell>
            <TableCell align="left">Value</TableCell>
            <TableCell align="left">Status and Interpretation</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          <TableRow key={Observation.id}>
            <TableCell align="left">{Observation.effectiveDateTime}</TableCell>
            <TableCell align="left">
              {generateObservableEntity(Observation)}
            </TableCell>
            <TableCell align="left">
              {generateObservableEntityValue(Observation)}
            </TableCell>
            <TableCell align="left">
              {generateObservableEntityStatusInterpretation(Observation)}
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </TableContainer>
  );
};
export default ObservationTable;
