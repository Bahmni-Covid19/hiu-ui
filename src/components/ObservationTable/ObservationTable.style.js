import styled from "styled-components";

const ObservationTableStyles = styled.div`
  .loader-container {
    display: flex;
    align-items: center;
    flex-direction: column;
  }
`;

export default ObservationTableStyles;
