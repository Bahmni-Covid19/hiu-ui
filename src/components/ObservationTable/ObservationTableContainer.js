function createData(type, key, value) {
  return { type, key, value };
}
let rows = [];
function generateObservationRows(Observation) {
  let keys = Object.keys(Observation);
  let values = Object.values(Observation);
  for (var i = 0; i < Object.keys(Observation).length; i++) {
    if (typeof values[i] === "string") {
      rows.push(createData("row", keys[i], values[i]));
    } else if (typeof values[i] === "object") {
      let childObject = values[i];
      generateObservationRows(childObject);
    }
  }
  return rows;
}

export default generateObservationRows;
