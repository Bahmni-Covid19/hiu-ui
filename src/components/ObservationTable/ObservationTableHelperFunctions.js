import React from "react";
import valueForObs from "./ObsValueHandlers";

function getConceptText(codeableConcept) {
  if (codeableConcept.text) {
    return codeableConcept.text;
  } else {
    if (codeableConcept.coding) {
      return codeableConcept.coding[0].display;
    }
  }
  return undefined;
}

function getConceptTextForComponents(components) {
  const componentControls = [];
  for (var i = 0; i < components.length; i++) {
    componentControls.push({
      text: getConceptText(components[i].code),
      value: valueForObs(components[i])
    });
  }
  return (
    <ul>
      {componentControls.map(c => (
        <li key={c.text}>
          {c.text} : {c.value}
        </li>
      ))}
    </ul>
  );
}

export const generateObservableEntity = observation => {
  return getConceptText(observation.code);
};

export const generateObservableEntityValue = observation => {
  return (
    <div>
      <div>{valueForObs(observation)}</div>
      {getConceptTextForComponents(observation.component)}
    </div>
  );
};

export const generateObservableEntityStatusInterpretation = observation => {
  return (
    <div>
      {observation.status ? <div>status: {observation.status}</div> : ""}
      {observation.interpretation ? (
        <div>
          interpretation: {getConceptText(observation.interpretation[0])}
        </div>
      ) : (
        ""
      )}
    </div>
  );
};
