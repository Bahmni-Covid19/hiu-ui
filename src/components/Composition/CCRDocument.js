import React from "react";

import DiagnosticReportComponent from "../../components/DiagnosticReport/DiagnosticReportComponent";
import ObservationTable from "../../components/ObservationTable/ObservationTable";

const CCRDocument = ({ consentReqId, data }) => {
  return (
    <div>
      CCR Document: number
      <ObservationTable data={data} />
      <DiagnosticReportComponent consentReqId={consentReqId} data={data} />
    </div>
  );
};

export default CCRDocument;
