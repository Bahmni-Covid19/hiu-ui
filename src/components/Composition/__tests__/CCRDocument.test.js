import React from "react";
import {shallow} from "enzyme";
import CCRDocument from "../CCRDocument";
import ObservationTable from "../../ObservationTable/ObservationTable";
import MedicationRequestsComponent from "../../Medication/MedicationRequestsComponent";

describe('CCRDocument', ()=>{
    const mockData= [
        {
            resourceType: "MedicationRequest",
            parentType:'Composition',
            status: "active",
            intent: "order",
            medicationReference: {reference: "Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2", display: "IV antibiotics"},
            subject: {display: "Shriya"},
            authoredOn: "2020-01-22",
            dosageInstruction: []
        },
        {
            resourceType: "Observation",
            parentType:'Composition',
            status: "active",
            intent: "order",
            medicationReference: {reference: "Medication/c2b8f3fd-6da0-44ed-bc91-8bb05ff614d2", display: "IV antibiotics"},
            subject: {display: "Shriya"},
            authoredOn: "2020-01-22",
            dosageInstruction: []
        }
    ];
    const wrapper = shallow(
        <CCRDocument
            consentReqId={'123'}
            compositionData={mockData}
        />
    );

    it('should render properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render observation table component with empty data', () => {
        expect(wrapper.find(ObservationTable).length).toEqual(1);
        expect(wrapper.find(ObservationTable).props().data).toEqual([]);
    });

    it('should render Medication Requests component', () => {
        expect(wrapper.find(MedicationRequestsComponent).length).toEqual(1);
        expect(wrapper.find(MedicationRequestsComponent).props().medicationRequests).toEqual([]);
    });
});