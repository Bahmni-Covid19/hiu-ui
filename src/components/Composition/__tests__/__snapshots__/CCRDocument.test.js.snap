// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CCRDocument should render properly 1`] = `
<div>
  Document type: TODO
  <ObservationTable
    data={Array []}
  />
  <MedicationRequestsComponent
    medicationRequests={Array []}
  />
</div>
`;
